### PR TITLE
Adding system tests for HappyBase Table.cells() and scan().

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

File: system_tests/bigtable_happybase.py [hunk 1]
```diff
@@ -406,6 +407,217 @@ def test_rows_with_timestamp(self):
...
+
+        # Limit the size of the scan
+        scan_result = list(table.scan(limit=1))
+        self.assertEqual(scan_result, [
+            (ROW_KEY1, row1_data),
+        ])
+
+        # Scan with a row prefix.
+        prefix = ROW_KEY2[:-1]
+        self.assertEqual(prefix, ROW_KEY3[:-1])
```
Comment (tseaver) [inline system_tests/bigtable_happybase.py]: Why make assertions about Python slicing of literal strings?

File: system_tests/bigtable_happybase.py [hunk 2]
```diff
@@ -406,6 +407,217 @@ def test_rows_with_timestamp(self):
...
+        scan_result_prefixed = list(table.scan(row_prefix=prefix))
+        self.assertEqual(scan_result_prefixed, [
+            (ROW_KEY2, row2_data),
+            (ROW_KEY3, row3_data),
+        ])
+
+        # Make sure our keys are sorted in order
+        row_keys = [ROW_KEY1, ROW_KEY2, ROW_KEY3]
+        self.assertEqual(row_keys, sorted(row_keys))
```
Comment (tseaver) [inline system_tests/bigtable_happybase.py]: Why make assertions about Python's sorting of literal strings?

**Comment by tseaver** [comment]
LGTM except for the extraneous assertions, which made me wonder if I were misreading something.

File: system_tests/bigtable_happybase.py (hunk 1 above)
Comment (dhermes) [inline system_tests/bigtable_happybase.py]: I'm making sure that both `ROW_KEY2` and `ROW_KEY3` start with the same prefix (i.e. `row-key2a` and `row-key2b` both start with `row-key2`).

I'm unclear on what you'd prefer

File: system_tests/bigtable_happybase.py (hunk 2 above)
Comment (dhermes) [inline system_tests/bigtable_happybase.py]: It's not about Python's sorting, but the lexical ordering of the string values (language independent). The backend will return results in key (lexical) order, so this is just a sanity check that the values we are using are in the order we expect. The results below depend on that order.

**Comment by dhermes** [comment]
LMK what you think of my comments. I'd like to get this merged and wrap up the Happybase system tests (AFAIK only one PR left for those targeting Cloud Bigtable backend).

File: system_tests/bigtable_happybase.py (hunk 1 above)
Comment (tseaver) [inline system_tests/bigtable_happybase.py]: Yeah, but they are defined as string literals at the top of the module, right?

File: system_tests/bigtable_happybase.py (hunk 2 above)
Comment (tseaver) [inline system_tests/bigtable_happybase.py]: Made a list of three literal strings, and then sorted them and compared the results.  That is the bit I don't get:  nothing in our code could cause that assertion to fail.

**Comment by tseaver** [comment]
You can merge as it stands, but maybe follow up with an explanation of why you think those assertions might fail.

**Comment by dhermes** [comment]
I don't think they'll fail. The assertions are there as a sanity check, i.e. we meant this to be true with the literals we created, but we want to make extra double sure.

**Pull request merged**